python/cudf/cudf/_lib/io/utils.pyx

```diff
@@ -67,3 +72,61 @@ cdef cppclass iobase_data_sink(data_sink):
 
     size_t bytes_written() with gil:
         return buf.tell()
+
+
+cdef class HostBuffer:
+    """ HostBuffer lets you spill cudf DataFrames from device memory to host
+    memory. Once in host memory the dataframe can either be re-loaded back
+    into gpu memory, or spilled to disk. This is designed to reduce the amount
+    of unnecessary host memory copies.
+
+    Examples
+    --------
+    .. code-block:: python
+      import shutil
+      import cudf
+
+      # read cudf DataFrame into buffer on host
+      df = cudf.DataFrame({'a': [1, 1, 2], 'b': [1, 2, 3]})
+      buffer = cudf.io.HostBuffer()
+      df.to_parquet(buffer)
+
+      # Copy HostBuffer back to DataFrame
+      cudf.read_parquet(df)
+
+      # Write HostBuffer to disk
+      shutil.copyfileobj(buffer, open("output.parquet", "wb"))
+    """
+    cdef vector[char] buf
+    cdef size_t pos
+
+    def __cinit__(self, int initial_capacity=0):
+        self.pos = 0
+        if initial_capacity:
+            self.buf.reserve(initial_capacity)
+
+    def __len__(self):
+        return self.buf.size()
+
+    def seek(self, size_t pos):
+        self.pos = pos
+
+    def tell(self):
+        return self.pos
+
+    def readall(self):
+        return self.read(-1)
+
+    def read(self, int n=-1):
+        if self.pos >= self.buf.size():
+            return b""
```

Comment:
It's a little bit awkward that this returns an empty bytes object while below returns a MemoryView. Can we return an empty memoryview?